google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionInstance.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.cloud.spanner.SessionClient.SessionId;
+import com.google.cloud.spanner.spi.v1.SpannerRpc;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.threeten.bp.Instant;
+
+/**
+ * A {@code Session} can be used to perform transactions that read and/or modify data in a Cloud
+ * Spanner database. Sessions are managed internally by the client library, and users need not be
+ * aware of the actual session management, pooling and handling.
+ */
+class SessionInstance {
```

Comment:
Maybe `SessionReference`? As in: This class is just a reference to a backend session.